ballerina/response_generator.bal

```diff
@@ -21,140 +21,175 @@ import graphql.parser;
+    returns anydata {
         if fieldNode.getName() == TYPE_NAME_FIELD {
             return getTypeNameFromValue(parentValue);
         }
         any fieldValue = parentValue.hasKey(fieldNode.getName()) ? parentValue.get(fieldNode.getName()): ();
         __Type fieldType = getFieldTypeFromParentType(self.fieldType, self.engine.getSchema().types, fieldNode);
-        Field 'field = new (fieldNode, fieldType, path = self.path, fieldValue = fieldValue);
-        self.context.resetInterceptorCount();
-        return self.engine.resolve(self.context, 'field);
+        Field 'field = new (fieldNode, fieldType, path = path.clone(), fieldValue = fieldValue);
+        Context context = self.context.cloneWithoutErrors();
+        context.resetInterceptorCount();
+        anydata result = self.engine.resolve(context, 'field);
+        self.context.addErrors(context.getErrors());
+        return result;
     }
 
-    isolated function getResultFromArray((any|error)[] parentValue, parser:FieldNode parentNode) returns anydata {
-        anydata[] result = [];
+    isolated function getResultFromArray((any|error)[] parentValue, parser:FieldNode parentNode, (string|int)[] path)
+    returns anydata {
+        if !self.isMutationOperation {
+            return self.getResultFromArrayParallelly(parentValue, parentNode, path);
+        }
         int i = 0;
+        anydata[] result = [];
         foreach any|error element in parentValue {
-            self.path.push(i);
-            anydata elementValue = self.getResult(element, parentNode);
+            path.push(i);
+            anydata elementValue = self.getResult(element, parentNode, path.clone());
+            i += 1;
+            _ = path.pop();
             if elementValue is ErrorDetail {
                 result.push(());
             } else {
                 result.push(elementValue);
             }
-            i += 1;
-            _ = self.path.pop();
         }
         return result;
     }
 
-    isolated function getResultFromTable(table<map<any>> parentValue, parser:FieldNode parentNode) returns anydata {
+    isolated function getResultFromArrayParallelly((any|error)[] parentValue, parser:FieldNode parentNode, (string|int)[] path)
+    returns anydata {
+        int i = 0;
         anydata[] result = [];
-        foreach any element in parentValue {
-            anydata elementValue = self.getResult(element, parentNode);
+        future<anydata>[] futures = [];
+        var getResult = self.getResult;
+        foreach any|error element in parentValue {
+            path.push(i);
+            future<anydata> 'future = start getResult(element, parentNode, path.clone());
+            futures.push('future);
+            i += 1;
+            _ = path.pop();
+        }
+        foreach future<anydata> elementFuture in futures {
+            anydata elementValue = checkpanic wait elementFuture;
```

Comment:
We should log the error and continue here as well.

Reply:
updated with [37c63d0](https://github.com/ballerina-platform/module-ballerina-graphql/pull/1256/commits/37c63d097d30bc515b70ab863f7e306118e3b2f7)